components/layout/global-header.tsx

```diff
@@ -1,47 +1,55 @@
-import { createElement, useRef, useState } from "react";
+import { createElement, useRef, useState, useEffect } from "react";
 
 import { Popover } from "@headlessui/react";
 import Link from "next/link";
 import clsx from "clsx";
+import { throttle } from "@martinstark/throttle-ts";
 import { Container } from "../ui/container";
 import { NavigationMenuPopover, navMenuButtonIcon } from "./navigation-menu";
 import { WBWLogoWhite } from "~/components/ui/wbw-logo";
 
-import useDocumentScrollThrottled from "~/lib/scroll-throttled";
+const MINIMUM_SCROLL = 200; // 64 height header
+const TIMEOUT_DELAY = 150;
 
 export function GlobalHeader() {
   const popoverButtonRef = useRef<HTMLButtonElement>(null);
 
   const [shouldHideHeader, setShouldHideHeader] = useState(false);
-  const [shouldShowShadow, setShouldShowShadow] = useState(false);
+  const [, setScrollPosition] = useState(0);
+  let previousScrollTop = 0;
 
-  const MINIMUM_SCROLL = 80;
-  const TIMEOUT_DELAY = 150;
+  function handleDocumentScroll() {
+    const { scrollTop: currentScrollTop } = document.documentElement;
+    const isScrolledDown = previousScrollTop < currentScrollTop;
+    const isMinimumScrolled = currentScrollTop > MINIMUM_SCROLL;
 
-  useDocumentScrollThrottled(
-    (callbackData: { previousScrollTop: Number; currentScrollTop: Number }) => {
-      const { previousScrollTop, currentScrollTop } = callbackData;
-      const isScrolledDown = previousScrollTop < currentScrollTop;
-      const isMinimumScrolled = currentScrollTop > MINIMUM_SCROLL;
+    setScrollPosition((previousPosition) => {
```

Comment:
We can not use same name for arguments and upper scope variables, it cause a confusing